### ISSUE TITLE
RFC: aliases via Cloudflare Worker

### ISSUE THREAD

**Issue by inyono** [serlo/serlo.org#272]
Two key-value stores per language:
- `DE_ALIAS_TO_ROUTE`: maps hashed route (e.g. hashed value of `mathe/funktionen/wichtige-funktionstypen-ihre-eigenschaften/quadratische-funktionen-parabeln/funktionsterm-quadratischen-funktion/parabel`) to route (e.g. `entity/view/1855`)
- `DE_ROUTE_TO_ALIAS`: maps route (e.g. `entity/view/1855`) to an object
  - current alias: `mathe/funktionen/wichtige-funktionstypen-ihre-eigenschaften/quadratische-funktionen-parabeln/funktionsterm-quadratischen-funktion/parabel`)
  - whether the alias may be purged

## Resolving aliases

- GET request to https://de.serlo.org/mathe/funktionen/wichtige-funktionstypen-ihre-eigenschaften/quadratische-funktionen-parabeln/funktionsterm-quadratischen-funktion/parabel
- Check in `DE_ALIAS_TO_ROUTE` whether the given path `mathe/funktionen/wichtige-funktionstypen-ihre-eigenschaften/quadratische-funktionen-parabeln/funktionsterm-quadratischen-funktion/parabel` has a route.
  - If yes, check via `DE_ROUTE_TO_ALIAS` whether there is a newer alias.
    - If not, proxy the path (e.g. `https://de.serlo.org/entity/view/1855`)
    - If yes, redirect to newest alias
  - If not, continue with usual request

## Setting aliases

-  Check in `DE_ALIAS_TO_ROUTE` whether the given alias is already in use.
  - If not, we have an alias.
  - If yes, check via `DE_ROUTE_TO_ALIAS` if the route is up-to-date.
    - If not, we can use that.
    - If it is up-to-date, generate new alias (e.g. by adding the id) and redo checks.
- When we found an alias, set that both in `DE_ALIAS_TO_ROUTE` and `DE_ROUTE_TO_ALIAS`

## Updating aliases

- Basically same as setting alias. Note that the old path still maps to the same route

## Trashing / restoring entities

- When trashing an entity, we mark the alias as purgable via `DE_ROUTE_TO_ALIAS`
- When restoring an entity, check if the current route is still free via `DE_ALIAS_TO_ROUTE`
  - If yes, mark the alias as non-purgeable
  - If not, we need to generate a new alias (see setting aliases)

## Open questions

- [ ] Backend routes? Should never be aliased

**Comment by inyono** [comment]
Resolved by #540 (needs some more testing before it goes to production).
